lang/compiler.go

```diff
@@ -280,14 +280,27 @@ func (p *Program) Start(ctx context.Context, alloc memory.Allocator) (flux.Query
 	// This span gets closed by the query when it is done.
 	s, cctx := opentracing.StartSpanFromContext(ctx, "execute")
 	results := make(chan flux.Result)
+
+	resourceAlloc, ok := alloc.(*memory.ResourceAllocator)
+	if !ok {
+		resourceAlloc = &memory.ResourceAllocator{
+			Allocator: alloc,
+		}
+	}
+
+	var statsAlloc memory.StatsAllocator
```

Comment:
This is the correct area to inject the GcAllocator, but I think it would be better to do this in a different way. We can define an extra field on the struct.

```
type query struct {
    alloc *memory.ResourceAllocator
    mem memory.Allocator
}
```

When we initialize this without the feature flag, they point to the same struct. When we enable the `GcAllocator`, `mem` will be pointed towards the version of `GcAllocator` that wraps another allocator.

Reply:
Seems risky since someone could accidentally use `alloc` thereby bypassing the wrapping allocator, but ok. Named it `allocatorStats` instead to try and make it clearer that it is not to be used as an allocator directly.

Reply:
Yea that's a good name.